packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-suggestions.tsx

```diff
@@ -91,6 +93,16 @@ export class AtomicSmartSnippetSuggestions implements InitializableComponent {
    */
   @Prop({reflect: true}) snippetStyle?: string;
 
+  @Listen('atomic/resolveResult')
+  public resolveResult(event: ResultContextEvent<Result>) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    this.smartSnippetQuestionsListState.questions.forEach((question) => {
+      if (question.source !== undefined) event.detail(question.source);
+    });
```

Comment:
This probably won't work all the time, since it's responding to 1 event multiple times

Reply:
I think we should listen on each suggestion separately to only provide them with the correct result. This could be done either by using `ref=` on each question or by creating a new component that would wrap the source.

Reply:
Will do !